compiler-core/src/ast/inlay_hints.rs

```diff
@@ -0,0 +1,104 @@
+use crate::{
+    ast::{
+        visit::{self, Visit},
+        SrcSpan, TypedAssignment, TypedExpr, TypedModule,
+    },
+    line_numbers::LineNumbers,
+    type_::pretty::Printer,
+};
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct InlayHint {
+    pub label: String,
+    pub offset: u32,
+}
+
+/// Determines if the expression is a simple literal whose inlayHints must not be showed
+/// in a pipeline chain
+fn is_simple_lit(expr: &TypedExpr) -> bool {
+    matches!(
```

Comment:
This can go on `TypedExpr`, but `matches` should not be used as it doesn't prompt the programmer to update this function when a new variant is added. This has repeatedly been a source of bugs.

Reply:
https://github.com/gleam-lang/gleam/pull/3290/commits/d141e9f2c24d6dc4bd35750dacc83bddb684cc92
👍